ZeCamp/ZeCamp/Sources/Schedule/ScheduleScreen.swift

```diff
@@ -86,24 +88,108 @@ struct TimeSlot {
 
 class ScheduleDelegate: NSObject, UITableViewDelegate {
     weak var viewController: UIViewController?
+    var timeSlots: [TimeSlot]
+    
+    init(_ timeSlots: [TimeSlot]){
+        self.timeSlots = timeSlots
+    }
     
     func tableView(_ tableView: UITableView, didSelectRowAt indexPath: IndexPath) {
-        let emptyScreen = UIViewController()
-        emptyScreen.title = "Event"
-        emptyScreen.view.backgroundColor = .white
+        defer {
+            tableView.deselectRow(at: indexPath, animated: true)
+        }
+        
+        
+        let id = timeSlots[indexPath.section].events[indexPath.row].id
+        print(id)
+        let eventScreen = EventScreen(id).getController();
+        
+        eventScreen.view.backgroundColor = .white
+        if #available(iOS 11.0, *) {
+            eventScreen.navigationItem.largeTitleDisplayMode = .never
+        }
+        viewController?.navigationController?.pushViewController(eventScreen, animated: true)
+    }
+    
+}
+
+class EventScreen {
+    
+    var eventId = 0
+    
+    init(_ id: Int) {
+        self.eventId = id
+    }
+    
+    func setupLabelConstraints(_ label: UILabel, _ eventScreen: UIViewController){
+        label.translatesAutoresizingMaskIntoConstraints = false
+        label.centerXAnchor.constraint(greaterThanOrEqualTo: eventScreen.view.centerXAnchor).isActive = true
+        label.widthAnchor.constraint(equalTo: eventScreen.view.widthAnchor, constant: -20).isActive = true
+        label.numberOfLines = 0
+    }
+    
+    func setDefaultStackView(_ stackView: UIStackView) {
```

Comment:
It's better to use pure functions, i.e. don't mutate the parameters. In this case you can create the `UIStackView` in this function and return it.